ingestion-sink/src/main/java/com/mozilla/telemetry/ingestion/util/KV.java

```diff
@@ -0,0 +1,19 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package com.mozilla.telemetry.ingestion.util;
+
+/**
+ * Container class for a key-value pair
+ */
+public class KV<KeyType, ValueType> {
+
+  public KeyType key;
+  public ValueType value;
+
+  public KV(KeyType key, ValueType value) {
```

Comment:
This class seems well suited to a private constructor and a public factory method so that usage is `KV.of(key, value)`.

Reply:
Yeah, I was having issues with a static method returning variable type because I haven't done that and couldn't get the syntax right, so I went the simple way here.

Do you think we should follow the beam approach in general of having private constructors and public factory methods everywhere?

Reply:
> Do you think we should follow the beam approach in general of having private constructors and public factory methods everywhere?

Yes, or even better AutoValue all the things, which eliminates some code duplication and enforces that pattern.